io/plugins/eu.esdihumboldt.hale.io.jdbc.msaccess.test/src/eu/esdihumboldt/hale/io/jdbc/msaccess/test/MsAccessDataReaderTestSuit.java

```diff
@@ -0,0 +1,250 @@
+		ByteSink dest = Files.asByteSink(getSourceTempFilePath());
+		source.copyTo(dest);
+	}
+
+	/**
+	 * Generates a random path (within the system's temporary folder) for the
+	 * source database.
+	 * 
+	 * @return the absolute path of the source temp file
+	 */
+	public File getSourceTempFilePath() {
+
+		if (TEMP_SOURCE_FILE_NAME == null) {
+			try {
+				TEMP_SOURCE_FILE_NAME = File.createTempFile(SOURCE_DB_NAME, SOURCE_DB_EXT);
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+		}
+		return TEMP_SOURCE_FILE_NAME;
+		// return getTempDir() + File.separator + getRandomNumber() + "_" +
+		// SOURCE_DB_NAME;
+	}
+
+	/**
+	 * Deletes the source temp file.
+	 */
+	public void deleteSourceTempFile() {
+		deleteTempFile(getSourceTempFilePath());
+	}
+
+	private void deleteTempFile(File tempFile) {
+		if (tempFile.exists()) {
+			tempFile.delete();
+		}
+	}
+
+	/**
+	 * To get {@link Connection} object by giving database location, user name
+	 * and password.
+	 * 
+	 * @return Connection object
+	 */
+	public Connection getConnection() {
+		Connection con = null;
+
+		try {
+			con = DriverManager.getConnection("jdbc:ucanaccess://" + getSourceTempFilePath(), USER_NAME, PASSWORD);
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+
+		return con;
+
+	}
+
+	/**
+	 * Connection to MS Access database and getting first row data.
+	 * 
+	 * @return String value, First row data joined by delimiter \t
+	 */
+	public String getFirstData() {
+		Connection con = null;
+		Statement st = null;
+		ResultSet rs = null;
+
+		try {
+			con = getConnection();
+
+			st = con.createStatement();
+			rs = st.executeQuery(SQL_QUERY);
+
+			StringBuilder row = new StringBuilder();
+			if (rs != null) {
+				rs.next();
+
+				row.append(String.valueOf(rs.getInt(1)));
+				row.append("\t");
+				row.append(rs.getString(2));
+				row.append("\t");
+				row.append(String.valueOf(rs.getInt(3)));
+				row.append("\t");
+				row.append(String.valueOf(rs.getInt(4)));
+
+			}
+
+			return row.toString();
+
+		} catch (Exception e) {
+			log.error("Could not able to get data from MsAccessDatabase", e);
+			return null;
+		} finally {
+			try {
+				if (st != null)
+					st.close();
+				if (con != null)
+					con.close();
+			} catch (SQLException e) {
+				// ignore
+			}
+		}
+	}
+
+	/**
+	 * Test - reads a sample MsAccess Database schema. UCanAccess lib should not
+	 * throw any error.
+	 * 
+	 * @throws Exception
+	 *             if an error occurs
+	 */
+	public void schemaReaderTest() throws Exception {
+		MsAccessSchemaReader schemaReader = new MsAccessSchemaReader();
+		schemaReader.setSource(new FileIOSupplier(getSourceTempFilePath()));
+		schemaReader.setParameter(JDBCSchemaReader.PARAM_USER, Value.of(USER_NAME));
+		schemaReader.setParameter(JDBCSchemaReader.PARAM_PASSWORD, Value.of(PASSWORD));
+		schemaReader.setIsSchemaNameQuoted(false);
```

Comment:
Setting this behavior should be done in your `MsAccessSchemaReader` class. After all you want to customize the behavior of the class, not make it adaptable by the user.